tests/lib/v9-rule-migration/v9-rule-migration.js

```diff
@@ -0,0 +1,286 @@
+                            const firstToken = sourceCode.getFirstToken(node);
+                            const firstTokens = sourceCode.getFirstTokens(node);
+                            const lastToken = sourceCode.getLastToken(node);
+                            const lastTokens = sourceCode.getLastTokens(node);
+                            const tokenAfter = sourceCode.getTokenAfter(node);
+                            const tokenBefore = sourceCode.getTokenBefore(node);
+                            const tokenByRangeStart = sourceCode.getTokenByRangeStart(node);
+                            const getTokens = sourceCode.getTokens(node);
+                            const tokensAfter = sourceCode.getTokensAfter(node);
+                            const tokensBefore = sourceCode.getTokensBefore(node);
+                            const tokensBetween = sourceCode.getTokensBetween(node);
+                            const parserServices = sourceCode.parserServices;
+                        },
+                    };
+                }
+            };
+            `.trim()
+            )
+        );
+    });
+
+    it("should migrate recently added methods on sourceCode with signature change", async () => {
+        const result = await applyTransform(
+            v9MigrationTransform,
+            `
+            module.exports = {
+                create(context) {
+                    return {
+                        Program(node) {
+                            const scope = context.getScope();
+                            const result = context.markVariableAsUsed("foo");
+                        },
+
+                        MemberExpression(memberExpressionNode) {
+                            const ancestor = context.getAncestors();
+                        },
+
+                        FunctionDeclaration(functionDeclarationNode) {
+                            const declaredVariables = context.getDeclaredVariables();
+                        },
+                    };
+                }
+            };
+            `
+        );
+
+        assert.strictEqual(
+            normalizeLineEndngs(result),
+            normalizeLineEndngs(
+                `
+            module.exports = {
+                create(context) {
+                    const sourceCode = context.sourceCode ?? context.getSourceCode();
+                    return {
+                        Program(node) {
+                            const scope = sourceCode.getScope ? sourceCode.getScope(node) : context.getScope();
+                            const result = sourceCode.markVariableAsUsed ? sourceCode.markVariableAsUsed("foo", node) : context.markVariableAsUsed();
+                        },
+
+                        MemberExpression(memberExpressionNode) {
+                            const ancestor = sourceCode.getAncestors ? sourceCode.getAncestors(memberExpressionNode) : context.getAncestors();
```

Comment:
Can you add a test for the situation when someone does something like this?

```js
const statements = context.getAncestors().filter(node => node.endsWith("Statement"));

Reply:
Added 👍🏻